libcudacxx/include/cuda/__functional/maximum.h

```diff
@@ -0,0 +1,59 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDA_FUNCTIONAL_MAXIMUM_H
+#define _CUDA_FUNCTIONAL_MAXIMUM_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__functional/binary_function.h>
+#include <cuda/std/__type_traits/common_type.h>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA
+
+template <class _Tp = void>
+struct _CCCL_TYPE_VISIBILITY_DEFAULT maximum : __binary_function<_Tp, _Tp, _Tp>
+{
+  using __result_type = _Tp; // used by valarray
+
```

Comment:
This can be dropped, we do not have valarry
```suggestion
```

Reply:
Note that for this (and binary_function), I mostly just copied from the existing code in `cuda/std/functional`. Those should probably also be similarly cleaned up

Reply:
Yeah this is one of the things we could remove from the existing ones. For the standard function objects we need the `__binary_function` inheritance for standard compliance in C++11 && C++14